### ISSUE TITLE
Auto-close server with Jupiter 5.1-M2

### ISSUE THREAD

**Issue by sormuras** [vert-x3/vertx-junit5#30]
Stored values that implement [ExtensionContext.Store.CloseableResource](https://github.com/junit-team/junit5/blob/215f6191fa484cedebba7cb76678a6d3bd9c00bc/junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/ExtensionContext.java#L295-L320) are notified when the enclosing extension context is closed.

Might simplify the "close vert.x server" after "usage" logic.

Here is a [simple demo](https://github.com/junit-team/junit5/blob/215f6191fa484cedebba7cb76678a6d3bd9c00bc/platform-tests/src/test/java/org/junit/jupiter/extensions/Heavyweight.java) that shows how to use this new feature.

**Comment by jponge** [comment]
Sounds good, fancy a PR? We could have it on the side and eventually merge it when 5.1 gets released.

**Comment by sormuras** [comment]
Hope to find some time to file a PR this week.

**Comment by jponge** [comment]
Great! You may want to check #34 then, as it features 3 types of closeable resources.

**Comment by sormuras** [comment]
Mh, as #34 is not yet merged and you know all 3 resources (by name, I guess) ... it should be less work if you just add `implements CloseableResource`. 😉

**Comment by sormuras** [comment]
When running `mvn test` I get 2 failures and 1 error. Same for master or the feature branch of #34. Surefire prints something like:

```
Failed tests: 
  VertxExtensionTest$EmbeddedWithARunner$FailureTest.thisMustFail org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
  VertxExtensionTest$EmbeddedWithARunner$FailureWithIntermediateAsyncVerifierTest.thisMustAlsoFail org.opentest4j.AssertionFailedError: expected: <58> but was: <69>
Tests in error: 
  VertxExtensionTest$EmbeddedWithARunner$DirectFailureTest.thisMustFail:152 Runtime

Tests run: 69, Failures: 2, Errors: 1, Skipped: 0
```

**Comment by jponge** [comment]
Java 9? We are still on the surefire version that doesn't properly exclude
tags.

**Comment by jponge** [comment]
Implemented in #34
